src/frontend/js/components/DashBoard/routes.tsx

```diff
@@ -0,0 +1,63 @@
+import { RouteObject } from 'react-router-dom';
+import { defineMessages, MessageDescriptor } from 'react-intl';
+
+export interface DashBoardRouteDefinition {
+  element: JSX.Element;
+  intlPath?: MessageDescriptor;
+  intlTitle: MessageDescriptor;
+  protected?: boolean;
+  show?: () => boolean;
+  title?: string;
+}
+
+export interface DashBoardRoute extends RouteObject {
+  path: string;
+  show?: () => boolean;
+  title?: string;
```

Comment:
Until `show` is explicitly sets to `false`, `title` should be mandatory as we need this information to display a link within the navigation.

```
export type DashBoardRoute = RouteObject & { path: string } & (
    | {
        show?: () => boolean;
        title: string;
      }
    | {
        show: () => false;
        title?: string;
      }
  );
```